### PR TITLE
Implemented registration tests

### PR DESCRIPTION
**Pull request by DanielZ818**

### CODE REVIEW

File: src/test/java/use_case/signup/SignupInteractorTest.java [hunk 1]
```diff
@@ -1 +1,181 @@
...
+        SignupInputData inputData = new SignupInputData(existingUser.getUserName(), existingUser.getPassword(), repeatedPassword);
+        SignupUserDataAccessObject signupDAO = new SignupUserDataAccessObject();
+        SignupInputBoundary interactor = new SignupInteractor(signupDAO, presenter, new UserFactory());
+        interactor.execute(inputData);
+
+        // Remove the test user
+        LoginSignupTestDataAccessObject deleteDAO = new LoginSignupTestDataAccessObject();
+        deleteDAO.delete(existingUser);
+    }
+
```
Comment (Huzaifa123451) [inline src/test/java/use_case/signup/SignupInteractorTest.java]: Good Comprehensive test cases - One potential improvement cold be to avoid duplicating presenter logic by introducing these following lines of codes as helper functions:

private LoginOutputBoundary createSuccessPresenter(User expected)
private LoginOutputBoundary createFailurePresenter(String expectedError)

This could make tests more readable and shorter.

**Review by mahirc27 — CHANGES_REQUESTED** [review]
Great job adding integration covrage for the Login and Signup usecases. The tests cover the happy paths, password mismatch, and edge cases like empty inputs effectively. However, there are a few issues regarding test stability that need to be looked at before merging.
In both LoginInteractorTest.java and SignupInteractorTest.java, the user deletion code runs at the end of the test.If an assertion fails (e.g., assertEquals throws error) or the interactor crashes the test exits, and the delete line is not reached. This leaves the test user in the database.Move the cleanup logic into a finally block or @AfterEach method to make sure it runs even if the test fails.

File: src/test/java/use_case/login/LoginInteractorTest.java
Comment (404Haze) [inline src/test/java/use_case/login/LoginInteractorTest.java]: You call your deleteDAO.delete(testUser) after your assertions in tests, so if one of your assertions fails, the cleanup never happens. Use @AfterEach with unique test usernames.

File: src/test/java/use_case/signup/SignupInteractorTest.java
Comment (404Haze) [inline src/test/java/use_case/signup/SignupInteractorTest.java]: your user instantiation has repeated field mapping, so instead of doing stuff like `params.put("level", String.valueOf(user.getLevel()))`, you can make a Builder pattern from the lecture notes, should make it easier to work with.

**Review by 404Haze — CHANGES_REQUESTED** [review]
Overall looks good, do take a read at the API key comment, need to make sure we don't leak anything.

File: src/main/java/frameworks_and_drivers/DataAccess/LoginSignupTestDataAccessObject.java
[the remark this answers is not in the document]
Comment (DanielZ818) [reply src/main/java/frameworks_and_drivers/DataAccess/LoginSignupTestDataAccessObject.java]: This API route is intended for testing only and cannot be used to delete other users, so keeping the key as a constant here is fine.

**Review by C20H12 — COMMENTED** [review]
missing test cases for unknown error

https://github.com/404Haze/CourseClash/blob/b289454dda1f5367b903222c141bbeb4d0d0ec35/src/main/java/use_case/registration/signup/SignupInteractor.java#L56

https://github.com/404Haze/CourseClash/blob/b289454dda1f5367b903222c141bbeb4d0d0ec35/src/main/java/use_case/registration/login/LoginInteractor.java#L51

**Review by 404Haze — APPROVED** [review]
Good job addressing my change requests, approved.

**Review by AlbertKang1004 — APPROVED** [review]
